test/test_entity.rb

```diff
@@ -104,6 +102,62 @@ def test_replace_entities
       assert_equal source, out
     end
 
+    def test_entity_with_reference
```

Comment:
Can we improve test name? This class is `EntityTester`. So `_entity` is redundant.

`test_readers_reference`?